pkg/watchdog/interface.go

```diff
@@ -17,6 +17,9 @@ type Watchdog interface {
 	GetTimeout() time.Duration
 	// LastFoodTime return the last time the watchdog was fed
 	LastFoodTime() time.Time
+	// Reset resets the watchdog
+	// This very probably should never be used in production code, but for tests only!
+	Reset()
```

Comment:
When possible, I think it's better not to update production code in order to accommodate test code.
This use case is a bit tricky, but we can probably wrap up the Fake dog with another layer that would have the `Reset` and `startFeeding` so those methods would be isolated in test code.
[Here](https://github.com/mshitrit/self-node/commit/e149b6975cf15bc2e5a9648977c00362848a04d0) is my attempt doing so.

Reply:
fair enough, and thanks for your suggestion with code. I took a slightly different approach though (see updated last commit), by

- using a new FakeWatchdog interface instead of struct, and extending the original fake impl struct, I think that's a bit clearer than having 2 fake watchdog types
- keeping startFeeding() in the sync impl for preventing code duplication

WDYT?

Reply:
looks great !
/lgtm